server/cmwell-data-tools/src/main/scala/cmwell/tools/data/downloader/consumer/BufferFillerActor.scala

```diff
@@ -230,7 +235,8 @@ class BufferFillerActor(threshold: Int,
       val to = toHint.map("&to-hint=" + _).getOrElse("")
 
       val uri =
-        s"${formatHost(baseUrl)}/$consumeHandler?position=$token&format=tsv$paramsValue$slowBulk$to"
+        s"${formatHost(baseUrl)}/$consumeHandler?position=$token&format=tsv$paramsValue$slowBulk$to" +
+          consumeLengthHint.fold(""){"&length-hint=" + _ }
```

Comment:
by using length-hint along with bulk-consume you are actually ruining the optimizations done by the server to save ES searches.
e.g. The amount of searches will be DOUBLED!

Reply:
The BufferFillerActor is not only used for STP also for all the data tools. Even if STP will be configured to use only consume the data tools still can use bulk consume.

Reply:
Ok - can get length-hint to run only for normal consume, no problem.
@YoniMataraso are you happy the length hint runs for all data tools, or do we need to work to limit it to STP only?